service/src/main/java/nlp/service/gate/service/GateNlpService.java

```diff
@@ -95,19 +99,68 @@ public NlpProcessingResult process(NlpInputPayload payload, Map<String, String>
     }
 
 
-    private GateApplicationParameters parseAppParams(ServiceConfiguration config) throws Exception {
+    /**
+     * Process a bulk request.
+     */
+    @Override
+    public List<NlpProcessingResult> processBulk(List<NlpInputPayload> payloads,
+                                                 Map<String, String> applicationParams) throws Exception {
+
+        // parse the payload to GenericDocument handler
+        //
+        List<GenericDocument> inputDocuments = new ArrayList<>();
+        for (NlpInputPayload singlePayload : payloads) {
+            GateNlpContentDataMapper contentMapper = new GateNlpContentDataMapper(singlePayload);
+
+            GenericDocument doc = new GenericDocument();
+            doc.setText(contentMapper.getText());
+
+            if (contentMapper.getAnnotations().size() > 0) {
+                doc.setAnnotations(contentMapper.getAnnotations());
+            }
+
+            // TODO: handle (if required):
+            // - document-level features
+            // - linked attributes
+            // - binary document
 
-        if (!config.getAppParams().containsKey(GateApplicationConfigurationKeys.GATE_HOME)) {
-            throw new Exception("GATE_HOME not set");
+            inputDocuments.add(doc);
         }
 
-        if (!config.getAppParams().containsKey(GateApplicationConfigurationKeys.GATE_APP_PATH)) {
-            throw new Exception("GATE_APP_PATH not set");
+        // run GATE processor
+        //
+        List<GenericDocument> outDocs;
+        try {
+            outDocs = gateProcessor.processDocumentsBulk(inputDocuments, applicationParams);
+        } catch (Exception e) {
+            String message = "Error processing NLP query: " + e.getMessage();
+            log.error(message);
+            throw e;
         }
 
-        GateApplicationParameters gateParams = new GateApplicationParameters();
+        // prepare the result payload
+        //
+        List<NlpProcessingResult> results = new ArrayList<>();
+        for (GenericDocument doc : outDocs) {
+            GateNlpResultDataMapper resultMapper = new GateNlpResultDataMapper();
+            resultMapper.setText(doc.getText());
+            resultMapper.setAnnotations(doc.getAnnotations());
+            resultMapper.setDocumentFeatures(doc.getDocumentFeatures());
+
+            // TODO: set in payload (if required):
+            // - binary document
+            // - enable/disable text inclusion
+
+            results.add(resultMapper.getProcessingResult());
```

Comment:
where is `getProcessingResult()`  set?

Reply:
`GateNlpResultDataMapper` is a proxy class, serving as an interface for creating objects of `NlpProcessingResult` type (could have actually used a builder pattern here) that will be returned as the result of NLP processing according to the API specs. `getProcessingResult()` returns the initialised object of `NlpProcessingResult` type, where the fields were set above by calling `resultMapper.set*()`.
The same applies for `GateNlpContentDataMapper` and `NlpInputPayload`.